### DISCUSSION TITLE
How to update a unique-value parameter?

### DISCUSSION THREAD

**Discussion by skonair** [sensedeep/dynamodb-onetable#413] [Q&A] [answered]
Hi folks,

I have a model which looks like this:
```
MyModel: {
      PK: { type: String, value: "${key}" },
      SK: { ... },
      key: { type: String },
      name: { type: String },
      uniqueName: { type: String, value: "${key}#${name}", unique: true },
      ...
```

Inserting works like a charm. The additional 'unique' item is created in the table. But when we update the `name` parameter of the inserted item, then the `name` and `uniqueName` parameters are changed in the `MyModel` item. But the 'unique' item is not changed and still has the old name in it.

I debugged the code (V2.5.0) and see that it works as implemented (of course). But I do not know what the correct way is to achieve a unique attribute with a combination of the PK and another parameter.

Any hint would be greatly appreciated!

Edit: Code reformatted

**Comment by mobsense** [comment, marked as answer]
Why are you providing your own value?  Unique should create its own unique item and the code is setup to remove the old unique values if you use the standard value template that OneTable uses internally.

If you still have trouble, could you create an issue and follow the issue template to submit a debug.ts that provides a test case for us to work on.

**Reply by skonair** [reply]
I wanted to use the unique attribute as an implementation for an SQL-like unique constraint on multiple columns. 
The `name` attribute should only be unique together with a specific PK `key`. 

I was totally missing out the `scope` attribute that I can use together with the `unique` attribute. This fully solves my problem.

```
MyModel: {
      PK: { type: String, value: "${key}" },
      SK: { ... },
      key: { type: String },
      name: { type: String, scope: "${key}", unique: true },
      ...
```
